library/std/src/sys/unix/kernel_copy.rs

```diff
@@ -114,11 +120,14 @@ impl FdMeta {
         }
     }
 
-    fn copy_file_range_candidate(&self) -> bool {
+    fn copy_file_range_candidate(&self, f: FdHandle) -> bool {
         match self {
             // copy_file_range will fail on empty procfs files. `read` can determine whether EOF has been reached
             // without extra cost and skip the write, thus there is no benefit in attempting copy_file_range
-            FdMeta::Metadata(meta) if meta.is_file() && meta.len() > 0 => true,
+            FdMeta::Metadata(meta) if f == FdHandle::Input && meta.is_file() && meta.len() > 0 => {
+                true
+            }
+            FdMeta::Metadata(meta) if f == FdHandle::Output && meta.is_file() => true,
             FdMeta::NoneObtained => true,
```

Comment:
Can you remove the NoneObtained case? It doesn't make sense anymore.

Reply:
Ah nice catch, I see that now too - I've force pushed with that removed (and squashed to one commit with accurate commit msg)